src/workerd/io/worker-interface.h

```diff
@@ -151,6 +151,105 @@ class WorkerInterface: public kj::HttpService {
 // for the promise, then invoke the destination object.
 kj::Own<WorkerInterface> newPromisedWorkerInterface(kj::Promise<kj::Own<WorkerInterface>> promise);
 
+template <typename Func>
+class LazyWorkerInterface final: public WorkerInterface {
+public:
+  LazyWorkerInterface(Func func): func(kj::mv(func)) {}
```

Comment:
Non-blocking nit: might consider marking this explicitly `KJ_DISALLOW_COPY_AND_MOVE` but not critical with having the KJ_ASSERT_NONNULLs

Reply:
Yea copies would be bad but they are implicitly disabled by virtue of having an Own member.
I don't see why moves should be a problem.

Reply:
Largely just an explicit hygiene thing for me... kind of a safeguard against someone changing any of the methods later and forgetting a `KJ_ASSERT_NONNULL`. But I'm being a bit extra paranoid :-)